src/orchestrator/orchestrator.rs

```diff
@@ -0,0 +1,281 @@
+    /// 6. After processing, executors are terminated to release resources.
+    ///
+    async fn create_aggregation_mst(
+        self,
+        executor_count: usize,
+    ) -> Result<AggregationMerkleSumTree<N_ASSETS, N_BYTES>, Box<dyn Error>>
+    where
+        [usize; N_ASSETS + 1]: Sized,
+        [usize; 2 * (1 + N_ASSETS)]: Sized,
+    {
+        let entries_per_executor = self.entry_csvs.len() / executor_count;
+
+        let mut executors = Vec::new();
+        let mut result_collectors = Vec::new();
+
+        let channel_size = std::env::var("CHANNEL_SIZE")
+            .unwrap_or_default()
+            .parse::<usize>()
+            .unwrap_or(32);
+
+        let cancel_token = CancellationToken::new();
+        let actual_number_of_workers = min(executor_count, self.entry_csvs.len());
+        for i in 0..actual_number_of_workers {
+            // Declare channels for communication
+            //
+            // There are three channels are used inthis method.
+            //
+            // - A `entries_tx` receives parsed data from the entry parser to distribute tasks to executors.
+            // - A `tree_tx` channel is used by the executors to send the results of the tasks.
+            // - A `stop` channel is used to send a stop signal to all executors in case of an error,
+            //   allowing for a coordinated shutdown or cancellation of tasks.
+            //
+            let (entries_tx, mut entries_rx) = mpsc::channel(channel_size);
+            let (tree_tx, tree_rx) = mpsc::channel(channel_size);
+
+            let executor = self.executor_spawner.spawn_executor().await;
+            result_collectors.push((i, tree_rx));
+
+            let cloned_cancel_token = cancel_token.clone();
+            executors.push(tokio::spawn(async move {
+                        loop {
+                            tokio::select! {
+                                entries_data = entries_rx.recv() => {
+                                    // When the distribution thread is finished, the channel will be closed.
+                                    let entries = match entries_data {
+                                        Some(entries) => entries,
+                                        None => break,
+                                    };
+                                    let processed_task = match executor.generate_tree::<N_ASSETS, N_BYTES>(entries).await {
+                                        Ok(entries) => entries,
+                                        Err(e) => {
+                                            eprintln!("Executor_{:?}: error while processing entries {:?}", i, e);
+                                            cloned_cancel_token.cancel();
+                                            break;
+                                        }
+                                    };
+                                    if tree_tx.send(processed_task).await.is_err() {
```

Comment:
Can you help me understanding this? Are you executing `generate_tree` here? Or what is `processed_task` ?

Reply:
In the data flow using channels, I named `task` somewhere in the orchestrator.
I little bit worrying gives you more confusion, but I can say the `processed_task` is a merkle sum tree generated by workers in runtime.